### PR TITLE
Add text about protected branches [fixes #160]

### LINKED ISSUE #160: (Discovery) enable protected branches in project repos
This is a best practice, but will require a little investigation to see what makes sense given the complex repo structure for this project.

### PR DESCRIPTION
**Pull request by woodt**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -235,3 +235,25 @@ reviewed with the team as part of the Sprint Review.
...
 - GitHub issue is marked Closed.
+
+
+## Additional Practices
+
+### Use of branch protection
+
+Branch protection affords the following advantages:
+
+- branch commit history can not be rewritten
```
Comment (adborden) [inline CONTRIBUTING.md]: cannot

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -235,3 +235,25 @@ reviewed with the team as part of the Sprint Review.
...
+- merges to protected branches can require a status check (e.g., passing tests, pull request reviews, etc.)
+
+Both GitHub and GitLab have support for protected branches:
+
+- https://help.github.com/en/github/administering-a-repository/configuring-protected-branches
+- https://docs.gitlab.com/ee/user/project/protected_branches.html
+
+#### Recommendations
+
+- Enable branch protection on master (or any deployable branch.)
```
Comment (adborden) [inline CONTRIBUTING.md]: - Enable require automated CI and security checks to pass before merging

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -235,3 +235,25 @@ reviewed with the team as part of the Sprint Review.
...
+- Enable "require pull request reviews before merging" on master
+- Enable commit signature verification
```
Comment (adborden) [inline CONTRIBUTING.md]: Do signed commits meet a security control? I'd rather not require folks to setup gpg signing unless it really helps us meet the controls.

Comment (woodt) [reply CONTRIBUTING.md]: I think this was a suggestion of @mogul ... I am happy to lose this, though.

Comment (mogul) [reply CONTRIBUTING.md]: Not required. Feel free to drop it and let an auditor bring it back if needed!

File: CONTRIBUTING.md (hunk 2 above)
Comment (woodt) [reply CONTRIBUTING.md]: Fixed (added to CONTRIBUTING.md)

**Comment by woodt** [comment]
"Ready to be reviewed/merged", he says!

**Pull request merged**